public/app/features/logs/components/panel/links.ts

```diff
@@ -0,0 +1,61 @@
+import { LinkModel } from '@grafana/data';
+
+import { FieldDef } from '../logParser';
+
+export function getTraceFromLinks(fields: FieldDef[]) {
+  for (const field of fields) {
+    if (!field.links) {
+      continue;
+    }
+    for (const link of field.links) {
+      const trace = getTraceFromLink(link);
+      if (trace) {
+        return trace;
+      }
+    }
+  }
+  return null;
+}
+
+function getTraceFromLink(link: LinkModel) {
+  const queryData = getDataSourceAndQueryFromLink(link);
+  if (!queryData || queryData.queryType !== 'traceql') {
+    return null;
+  }
+  return queryData;
+}
+
+export type EmbeddedInternalLink = {
+  dsUID: string;
+  query: string;
+  queryType: string;
+};
+
+function getDataSourceAndQueryFromLink(link: LinkModel): EmbeddedInternalLink | null {
```

Comment:
I faced the same problem when exploring other ways to use data links and thought instead of re-parsing the URL (which may be fragile) you could preserve interpolated query in the model is it can be re-used like `linkModel.meta.interpolated.query` (https://github.com/grafana/grafana/compare/ifrost/custom-data-links-behavior?expand=1)